### PR TITLE
add(main): libvgm

### LINKED ISSUE #13888: [Package]: libvgm
### Why do you want to add this package to Termux?

This is required for vgm2flac to work. I need vgm2flac to convert VGZ files to flac. ffmpeg can only convert the first 5 seconds of vgz files and vgmstream_cli can't convert vgz files even though vgmstream_cli is also a required dependency for vgm2flac to work.

### Home page URL

https://github.com/ValleyBell/libvgm

### Source code URL

https://github.com/ValleyBell/libvgm

### Packaging policy acknowledgement

- [X] The requested package complies with [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by 2096779623**
Close #13888 .

### CODE REVIEW

File: packages/libvgm/pthread_cancel.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+@@ -78,8 +78,11 @@
+ 	
+ 	if (! thr->id)
+ 		return;
+-	
++#ifndef __ANDROID__	
+ 	retVal = pthread_cancel(thr->id);
++#else
++	retVal = pthread_kill(thr->id, SIGUSR2);
++#endif
```
Comment (xtkoba) [inline packages/libvgm/pthread_cancel.patch:13]: This is not correct.

Comment (xtkoba) [reply packages/libvgm/pthread_cancel.patch:13]: More precisely, this does not work as expected unless the thread changes the action for `SIGUSR2` by using `sigaction(2)`.

Comment (2096779623) [reply packages/libvgm/pthread_cancel.patch:13]: So should I use `sig`?

Comment (xtkoba) [reply packages/libvgm/pthread_cancel.patch:13]: I would say you are not qualified to package this, unfortunately.

Comment (Grimler91) [reply packages/libvgm/pthread_cancel.patch:13]: To elaborate I guess @xtkoba means that a signal-catching function needs to be set for SIGUSR2, as done for example in [@xtkoba's calcurse patch](https://github.com/termux/termux-packages/blob/master/packages/calcurse/threading.patch#L7-L29)

**Pull request closed without merging**